tagging_accuracy/tagging_accuracy_report.py

```diff
@@ -108,7 +115,54 @@ def match_tags(self):
             f.write(json.dumps(no_tags_list))
         with open('partially_correct.json', 'w') as pc_json:
             pc_json.write(json.dumps(partially_correct_list, indent=4, sort_keys=True))
-        return no_tags, man_tags_not_in_result, man_tags_not_collected, total_packages, correctly_tagged, partially_correct
+        return (no_tags, man_tags_not_in_result, man_tags_not_collected, total_packages,
+                correctly_tagged, partially_correct)
+
+    def check_tag_matches(self, num_tags_in_result=4):
+        """Check the common tags between two packages and aggregate metric.
+
+        This metric is of importance from the point of view of the PGM as for us to give better
+        alternate and companion package recommendations we need that our packages be grouped
+        under common topics.
+        """
+        occurences = {}
+        # We need to compare each package to every other package in order to make sure that we
+        # cover all bases. At the same time we need to make sure that we don't consider any
+        # combo twice.
+        automated_tags_dict = self.automated_tags_dict
+        for package_name in list(automated_tags_dict):
```

Comment:
I guess slicing makes it more readable, I used list() because of the same reason that you mentioned - I can't use the iterator returned by .keys() as I'm modifying the dict.

Reply:
Apparently this doesn't work in Python 3.6 🙂 

```python
>>> d = {'a': '2', 'b': '3'}
>>> d.keys()
dict_keys(['a', 'b'])
>>> d.keys()[:]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'dict_keys' object is not subscriptable
>>>
```

Reply:
Needs to be casted to `list` explicitly. In that case this is probably not needed. 

Anyway, using `keys()` could improve readability.

Reply:
Right, makes sense.